conjure_oxide/src/rules/minion.rs

```diff
@@ -182,14 +216,71 @@ fn geq_to_ineq(expr: &Expr) -> Result<Expr, RuleApplicationError> {
 * ```
 */
 #[register_rule(("Minion", 100))]
-fn leq_to_ineq(expr: &Expr) -> Result<Expr, RuleApplicationError> {
+fn leq_to_ineq(expr: &Expr, _: &Model) -> ApplicationResult {
     match expr {
-        Expr::Leq(metadata, a, b) => Ok(Expr::Ineq(
+        Expr::Leq(metadata, a, b) => Ok(Reduction::pure(Expr::Ineq(
             metadata.clone(),
             a.clone(),
             b.clone(),
             Box::new(Expr::Constant(Metadata::new(), Const::Int(0))),
-        )),
-        _ => Err(RuleApplicationError::RuleNotApplicable),
+        ))),
+        _ => Err(ApplicationError::RuleNotApplicable),
     }
 }
+
+// #[register_rule(("Minion", 100))]
+// fn safediv_eq_to_diveq(expr: &Expr, _: &Model) -> ApplicationResult {
+//     match expr {
+//         Expr::Eq(metadata, a, b) => {
+//             if let Expr::SafeDiv(_, x, y) = a.as_ref() {
+//                 if !(b.is_reference() || b.is_constant()) {
+//                     return Err(ApplicationError::RuleNotApplicable);
+//                 }
+//                 Ok(Reduction::pure(Expr::DivEq(
+//                     metadata.clone(),
+//                     x.clone(),
+//                     y.clone(),
+//                     b.clone(),
+//                 )))
+//             } else if let Expr::SafeDiv(_, x, y) = b.as_ref() {
+//                 if !(a.is_reference() || a.is_constant()) {
+//                     return Err(ApplicationError::RuleNotApplicable);
+//                 }
+//                 Ok(Reduction::pure(Expr::DivEq(
+//                     metadata.clone(),
+//                     x.clone(),
+//                     y.clone(),
+//                     a.clone(),
+//                 )))
+//             } else {
+//                 Err(ApplicationError::RuleNotApplicable)
+//             }
+//         }
+//         _ => Err(ApplicationError::RuleNotApplicable),
+//     }
+// }
+
+#[register_rule(("Minion", 100))]
```

Comment:
not sure if we want to rewrite `a!=b` ~~> `alldiff(a,b)`

we would want to rewrite a clique of neqs (*) to an alldiff, but that's a bit more complex than what you have here.

(*) `a!=b /\ a!=c /\ b!=c` ~~> `allDiff([a,b,c])`

Reply:
What could we rewrite it to instead? @ozgurakgun

Reply:
I will resolve this in a later PR!